src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerContext.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.ComponentModel;
+using System.Diagnostics;
+
+namespace System.Text.Json.Serialization.Metadata
+{
+    /// <summary>
+    /// Provides metadata about a set of types that is relevant to JSON serialization.
+    /// </summary>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public abstract partial class JsonSerializerContext
+    {
+        internal JsonSerializerOptions? _options;
+
+        /// <summary>
+        /// Gets the run-time specified options of the context.
+        /// </summary>
+        /// <remarks>
+        /// The instance cannot be mutated once it is bounded with the context instance.
+        /// </remarks>
+        public JsonSerializerOptions Options
+        {
+            get
+            {
+                Debug.Assert(_options != null);
+                return _options;
+            }
+        }
+
+        /// <summary>
+        /// Creates an instance of <see cref="JsonSerializerContext"/> and binds it with the indicated <see cref="JsonSerializerOptions"/>.
+        /// If no options are provided, then a new instance with default options is created and bound.
+        /// </summary>
+        /// <param name="options">The run-time provided options for the context instance.</param>
+        protected JsonSerializerContext(JsonSerializerOptions? options)
+        {
+            if (options == null)
+            {
+                options = new JsonSerializerOptions();
```

Comment:
Nit: `options ??= new JsonSerializaterOptions();`

Reply:
I wrote L39-46 this way so that the `null` check for `options._context` will not be done when we create a brand new options instance.